### PR TITLE
Feature/spm integration

### PR DESCRIPTION
**Pull request by bnsports**

### CODE REVIEW

File: fearless/Common/Extension/Model/CryptoType+Utils.swift [hunk 1]
```diff
@@ -1,8 +1,8 @@
 import Foundation
-import SSFUtils
+import SSFModels
 
 extension CryptoType {
-    init(_ utilsType: SSFUtils.CryptoType) {
+    init(_ utilsType: CryptoType) {
```
Comment (DRadmir) [inline fearless/Common/Extension/Model/CryptoType+Utils.swift:5]: please check this, looks as init from self

File: fearless/Common/Extension/Model/CryptoType+Utils.swift [hunk 2]
```diff
@@ -13,7 +13,7 @@ extension CryptoType {
         }
     }
 
-    var utilsType: SSFUtils.CryptoType {
+    var utilsType: CryptoType {
```
Comment (DRadmir) [inline fearless/Common/Extension/Model/CryptoType+Utils.swift:16]: please check this property, look as useless

File: fearless/Common/Model/AccountItem.swift [hunk 3]
```diff
@@ -1,34 +1,35 @@
...
-        case "ed25519":
-            self = .ed25519
-        case "ecdsa":
-            self = .ecdsa
-        default:
-            return nil
-        }
-    }
-}
+// enum CryptoType: UInt8, Codable, CaseIterable {
```
Comment (DRadmir) [inline fearless/Common/Model/AccountItem.swift:4]: comments

File: fearless/CoreLayer/ProviderFactory/HistoryDataProviderFactory.swift [hunk 4]
```diff
@@ -21,12 +21,12 @@ class HistoryDataProviderFactory: BaseDataProviderFactory, HistoryDataProviderFa
     let operationFactory: HistoryOperationFactoryProtocol
 
     init(
-        cacheFacade: StorageFacadeProtocol,
+        cacheFacade _: StorageFacadeProtocol,
```
Comment (DRadmir) [inline fearless/CoreLayer/ProviderFactory/HistoryDataProviderFactory.swift:24]: unused

File: fearless/Modules/BackupCreatePassword/BackupCreatePasswordInteractor.swift [hunk 5]
```diff
@@ -197,20 +198,17 @@ final class BackupCreatePasswordInteractor: BaseAccountConfirmInteractor {
     private func saveBackupAccountToCloudStorage(
         account: OpenBackupAccount,
         password: String,
-        wallet: MetaAccountModel
+        wallet _: MetaAccountModel
```
Comment (DRadmir) [inline fearless/Modules/BackupCreatePassword/BackupCreatePasswordInteractor.swift:201]: you missed line with `self?.didBackuped(wallet: wallet)`

File: fearless/Modules/BackupSelectWallet/BackupSelectWalletInteractor.swift [hunk 6]
```diff
@@ -24,7 +24,7 @@ extension BackupSelectWalletInteractor: BackupSelectWalletInteractorInput {
             do {
                 if let cloudStorageService = cloudStorageService {
                     cloudStorageService.disconnect()
-                    let accounts = try await cloudStorageService.getFearlessBackupAccounts()
+                    let accounts = try await cloudStorageService.getBackupAccounts()
```
Comment (DRadmir) [inline fearless/Modules/BackupSelectWallet/BackupSelectWalletInteractor.swift:27]: please make sure what `getBackupAccounts()` it's right method

File: fearless/Modules/BackupWallet/BackupWalletInteractor.swift [hunk 7]
```diff
@@ -15,7 +15,7 @@ protocol BackupWalletInteractorOutput: AnyObject {
 }
 
 final class BackupWalletInteractor {
-    var cloudStorage: FearlessCompatibilityProtocol?
+    var cloudStorage: CloudStorageServiceProtocol?
```
Comment (DRadmir) [inline fearless/Modules/BackupWallet/BackupWalletInteractor.swift:18]: CloudStorageServiceProtocol provides fearless compatibility (mobile web)?

File: fearless/Modules/BackupWallet/BackupWalletInteractor.swift [hunk 8]
```diff
@@ -85,7 +85,7 @@ final class BackupWalletInteractor {
         Task {
             do {
                 if let cloudStorage = cloudStorage {
-                    let accounts = try await cloudStorage.getFearlessBackupAccounts()
+                    let accounts = try await cloudStorage.getBackupAccounts()
```
Comment (DRadmir) [inline fearless/Modules/BackupWallet/BackupWalletInteractor.swift:88]: please make sure what `cloudStorage.getBackupAccounts()` is right method

File: fearless/Modules/BackupWalletImported/BackupWalletImportedInteractor.swift [hunk 9]
```diff
@@ -5,7 +5,7 @@ import SSFCloudStorage
 protocol BackupWalletImportedInteractorOutput: AnyObject {}
 
 final class BackupWalletImportedInteractor {
-    var cloudStorageService: FearlessCompatibilityProtocol?
+    var cloudStorageService: CloudStorageServiceProtocol?
```
Comment (DRadmir) [inline fearless/Modules/BackupWalletImported/BackupWalletImportedInteractor.swift:8]: please make sure what CloudStorageServiceProtocol provides right method for fearless compatibility (mobile web) backups

File: fearless/Modules/CrossChainConfirmation/CrossChainDepsContainer.swift [hunk 10]
```diff
@@ -77,7 +77,7 @@ final class CrossChainDepsContainer {
 
         let fromChainData = XcmAssembly.FromChainData(
             chainId: originalChainAsset.chain.chainId,
-            cryptoType: SFCryptoType(utilsType: cryptoType.utilsType, isEthereum: response.isEthereumBased),
+            cryptoType: SFCryptoType(utilsType: cryptoType, isEthereum: response.isEthereumBased),
```
Comment (DRadmir) [inline fearless/Modules/CrossChainConfirmation/CrossChainDepsContainer.swift:80]: Can we use single CryptoType? Perhaps possible to remove SFCryptoType

File: fearless/Modules/Crowdloan/CrowdloanList/CrowdloanListInteractor.swift [hunk 11]
```diff
@@ -160,7 +160,7 @@ final class CrowdloanListInteractor: RuntimeConstantFetching {
             return
         }
 
-        displayInfoProvider = jsonDataProviderFactory.getJson(for: crowdloanUrl)
+        displayInfoProvider = try? jsonDataProviderFactory.getJson(for: crowdloanUrl)
```
Comment (DRadmir) [inline fearless/Modules/Crowdloan/CrowdloanList/CrowdloanListInteractor.swift:163]: I propose implement do catch block inside jsonDataProviderFactory

File: fearless/Modules/OnbordingMain/OnboardingMainInteractor.swift [hunk 12]
```diff
@@ -6,13 +6,13 @@ final class OnboardingMainInteractor {
     weak var presenter: OnboardingMainInteractorOutputProtocol?
 
     private let keystoreImportService: KeystoreImportServiceProtocol
-    private let cloudStorage: FearlessCompatibilityProtocol
+    private let cloudStorage: CloudStorageServiceProtocol
```
Comment (DRadmir) [inline fearless/Modules/OnbordingMain/OnboardingMainInteractor.swift:9]: please make sure what this right service for fearless

File: fearless/Modules/OnbordingMain/OnboardingMainInteractor.swift [hunk 13]
```diff
@@ -54,7 +54,7 @@ extension OnboardingMainInteractor: OnboardingMainInteractorInputProtocol {
         Task {
             do {
                 cloudStorage.disconnect()
-                let accounts = try await cloudStorage.getFearlessBackupAccounts()
+                let accounts = try await cloudStorage.getBackupAccounts()
```
Comment (DRadmir) [inline fearless/Modules/OnbordingMain/OnboardingMainInteractor.swift:57]: please make sure what this right method

**Review by DRadmir — CHANGES_REQUESTED** [review]

File: fearless/Common/DataProvider/PriceProviderFactory.swift [hunk 14]
```diff
@@ -1,9 +1,10 @@
 import Foundation
 import RobinHood
 import SSFModels
+import SSFSingleValueCache
 
 protocol PriceProviderFactoryProtocol {
-    func getPricesProvider(currencies: [Currency]?) -> AnySingleValueProvider<[PriceData]>
+    func getPricesProvider(currencies: [Currency]?) throws -> AnySingleValueProvider<[PriceData]>
```
Comment (DRadmir) [inline fearless/Common/DataProvider/PriceProviderFactory.swift:7]: I suggest implement do catch block inside PriceProviderFactory with LoggerProtocol

**Review by DRadmir — CHANGES_REQUESTED** [review]

File: fearless/Modules/BackupSelectWallet/BackupSelectWalletInteractor.swift (hunk 6 above)
Comment (bnsports) [reply fearless/Modules/BackupSelectWallet/BackupSelectWalletInteractor.swift:27]: let's test

**Review by DRadmir — APPROVED** [review]

**Pull request merged**